### ISSUE TITLE
Extending expiration on key with subkey means subkey stays expired

### ISSUE THREAD

**Issue by Jerbell** [vanitasvitae/pgpainless#420]
Hi. I create a key, primary is certify/sign, and has a subkey with encrypt. This key has now expired, and I use your method call:
`PGPainless.modifyKeyRing(this).setExpirationDate(expirationDate, protector).done()` to extend the expiration.
This will change the expiration of the primary key, but not the subkey. This is intended according to the documentation, so I was just wondering what the best way of extending the expiration of the subkey would be? Or is it best to generate a new subkey? 
Thanks!

Edit: maybe having a Pgpainless method to set the expiration on all keys which are not revoked would be better?

**Comment by vanitasvitae** [comment]
Huh, you are right, this functionality is currently missing 😮 

I will create a patch adding a method to align the expiration date of all non-revoked keys to a given date, as well as a method to set the expiration for a specific subkey.

**Comment by vanitasvitae** [comment]
You can now do
```java
secretKeys = PGPainless.modifyKeyRing(secretKeys)
    .setExpirationDateOfSubkey(expiration, subkeyId, secretKeyRingProtector)
    .done()
```

to change the expiration of a subkey.

The code is currently living in the `release/1.6` branch only. Could you check that out and see if it fixes your issues?
If so, I will try to release 1.6.5 soon and port the fix to the main branch.

**Comment by Jerbell** [comment]
Thanks - will try this soon.

**Comment by Jerbell** [comment]
It looks like you could probably rename it - since it seems as though it covers the primary key id as well?  I'm still trying it out :)

**Comment by Jerbell** [comment]
Yep - seems to work :) I'd probably just rename it, but up to you. All good!

**Comment by vanitasvitae** [comment]
> It looks like you could probably rename it - since it seems as though it covers the primary key id as well?

If you pass it the id of the primary key, it will simply call the old method to set the expiration date for the primary key and primary user id.

But if you pass it a subkey id instead, it will simply set the expiration date on the subkey binding signature, so only for the subkey itself.

**Comment by vanitasvitae** [comment]
Sorry for taking this long, family stuff got in the way.
I just released PGPainless 1.6.5 which contains the new methods :)

**Comment by Jerbell** [comment]
Thanks Sir Paul!
